### PR TITLE
docs: update dependency injection examples

### PR DESCRIPTION
**Pull request by emonddr**
Update dependency injection examples that mention authentication

Related to : https://github.com/strongloop/loopback-next/issues/2940

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: docs/site/Dependency-injection.md [hunk 1]
```diff
@@ -62,24 +93,27 @@ In LoopBack, we use [Context](Context.md) to keep track of all injectable
...
 
-function loginUser(username, password, cb) {
-  // check that username + password are valid
-}
+server
+  .bind(JWTAuthenticationStrategyBindings.TOKEN_EXPIRES_IN)
+  .to('600');
 ```
 
 However, when you want to create a binding that will instantiate a class and
```
Comment (jannyHou) [inline docs/site/Dependency-injection.md]: @emonddr One more thing to update here:

However, when you want to create a binding that will instantiate a configured class(as a provider) and automatically inject required dependencies, then you need to use .toProvider() method:

```ts
server
  .bind(AuthenticationBindings.AUTH_ACTION)
  .toProvider(AuthenticateActionProvider);

const provider = await server.get(AuthenticationBindings.AUTH_ACTION);
```

**Comment by raymondfeng** [comment]
In the longer term, I wonder if we should use simpler examples at https://github.com/strongloop/loopback-next/tree/master/examples/context for the DI docs.

**Comment by emonddr** [comment]
@raymondfeng 

> In the longer term, I wonder if we should use simpler examples at https://github.com/strongloop/loopback-next/tree/master/examples/context for the DI docs.

Hi Raymond, with the examples I have in place at the moment of my 2nd commit, are you satisfied with the 3 different examples (for the purposes of the authentication documentation changes), and want to create a separate task for modifying these dependency injection examples to be done at a later time?  Or are you saying that my PR should use non-authentication examples right now?
Please advise.

**Comment by raymondfeng** [comment]
> Hi Raymond, with the examples I have in place at the moment of my 2nd commit, are you satisfied with the 3 different examples (for the purposes of the authentication documentation changes), and want to create a separate task for modifying these dependency injection examples to be done at a later time? Or are you saying that my PR should use non-authentication examples right now?
Please advise.

I'm fine to use authentication examples for now. It would be ideal to use simpler standalone examples in the future PRs.

**Comment by emonddr** [comment]
@raymondfeng 

> I'm fine to use authentication examples for now. It would be ideal to use simpler standalone examples in the future PRs. 

Me too. Ok thanks. Can you approve it then?  And then I will rebase, etc. Thanks :)

**Pull request merged**